esp32_ulp/preprocess.py

```diff
@@ -42,12 +82,33 @@ def expand_defines(self, line):
 
         return line
 
+    def expand_rtc_macros(self, line):
+        clean = line.strip()
+        if len(clean) == 0:
+            return line
+
+        macro = clean.split('(', 1)
+        if len(macro) != 2:
+            return line
+
+        expand_fn = getattr(RTC_Macros, macro[0], None)
+        if expand_fn is None:
+            return line
+
+        _, args = macro
+        args, _ = args.rsplit(')', 1)  # trim away right bracket. safe because comments should already be stripped
+        args = args.split(',')  # not safe when string args contain ',' but we should not have those
+        args = [x.strip() for x in args]
+
+        return expand_fn(args)
```

Comment:
instead of giving a args list to the fn, you could give *args - and then you could save that args-splitting line in these function bodies by just having separate args instead of the list in the fn header.

Reply:
Very nice! Just did that and looks much cleaner.